src/main/sensors/battery.c

```diff
@@ -270,42 +269,34 @@ void updateCurrentMeter(int32_t lastUpdateAt, rxConfig_t *rxConfig, uint16_t dea
 
             updateCurrentDrawn(lastUpdateAt);
 
-            updateConsumptionWarning();
-
             break;
         case CURRENT_SENSOR_VIRTUAL:
-            amperageLatest = (int32_t)batteryConfig->currentMeterOffset;
+            amperage = (int32_t)batteryConfig->currentMeterOffset;
             if (ARMING_FLAG(ARMED)) {
                 throttleStatus_e throttleStatus = calculateThrottleStatus(rxConfig, deadband3d_throttle);
                 int throttleOffset = (int32_t)rcCommand[THROTTLE] - 1000;
                 if (throttleStatus == THROTTLE_LOW && feature(FEATURE_MOTOR_STOP)) {
                     throttleOffset = 0;
                 }
                 int throttleFactor = throttleOffset + (throttleOffset * throttleOffset / 50);
-                amperageLatest += throttleFactor * (int32_t)batteryConfig->currentMeterScale  / 1000;
+                amperage += throttleFactor * (int32_t)batteryConfig->currentMeterScale  / 1000;
             }
-            amperage = amperageLatest;
 
             updateCurrentDrawn(lastUpdateAt);
 
-            updateConsumptionWarning();
-
             break;
         case CURRENT_SENSOR_ESC:
-            #ifdef USE_ESC_TELEMETRY
-            if (isEscTelemetryActive()) {
-                amperageLatest = getEscTelemetryCurrent();
-                amperage = amperageLatest;
-                mAhDrawn = getEscTelemetryConsumption();
-
-                updateConsumptionWarning();
+            #ifdef USE_ESC_SENSOR
+            if (feature(FEATURE_ESC_SENSOR))
+            {
+                amperage = getEscSensorCurrent();
+                mAhDrawn = getEscSensorConsumption();
             }
+            #endif
```

Comment:
This should be after the `break`, to make it properly fall through if `USE_ESC_SENSOR` is not defined.